docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -338,19 +338,182 @@ The result will look like this:
 Hello, world
 ```
 
+### Directories and files
+
+It is possible to automatically load a filesystem path as a `Directory` or `File` object in a Dagger Function, by passing it as a default context to the corresponding argument. The `Directory` or `File` loaded in this manner is not merely a string, but it is the actual filesystem state of the specified directory or file, managed by the Dagger Engine and handled in code just like any another variable.
+
+:::important
+Default contexts are only available for arguments of type `Directory` and `File`. They are commonly used to load constant filesystem locations, such as an application's source code directory.
+:::
+
+The default context is set by adding a `DefaultPath` pragma (Go), `DefaultPath` annotation (Python), or `defaultPath` decorator (TypeScript) on the corresponding Dagger Function argument.
+
+The default context is resolved depending on various factors:
+- If an absolute path is specified:
+   - in a Git repository (defined by the presence of a `.git` sub-directory), the default context is the root of the Git repository.
+   - in a non-repository location (defined by the absence of a `.git` sub-directory), the default context is the directory containing a `dagger.json` file.
+- If a relative path is specified
+  - the default context is always the directory containing a `dagger.json` file.
```

Comment:
Should this instead be "the directory containing the module source code"? See https://github.com/dagger/dagger/pull/8339/files#r1754490348

Reply:
No, we fixed it https://github.com/dagger/dagger/pull/8391